lib/scrabble/scrabble.rb

```diff
@@ -0,0 +1,119 @@
+module ScrabbleSinatra
+  class ScrabbleSinatra
+    VALUES = {a: 1, e: 1, i: 1, o: 1, u: 1, l: 1, n: 1, r: 1, s: 1, t: 1,
+      d: 2, g: 2,
+      b: 3, c: 3, m: 3, p: 3,
+      f: 4, h: 4, v: 4, w: 4, y: 4,
+      k: 5,
+      j: 8, x: 8,
+      q: 10, z: 10
+    }
+
+    # this returns the score for each letter, total score for the word, and labels winning word
+    def self.scoring(word_array, score_method)
+      display_array = []
+      # checks if there is more than one word entered; used to label winning word
+      winning_word = highest_score_from(word_array, score_method) if word_array.length > 1
+      word_array.each do |word|
+        letter_hash = score_multiple(word.split(//))
+        letter_hash.each do |letter, score|
+          display_array << "#{letter}: #{score}"
+        end
+        word_score = "#{word}: #{score(word)}"
+        # if winning_word is undefined then word not labeled
+        word_score += "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; This one wins!" if word == winning_word
+        display_array << word_score
+      end
+      return display_array
+    end
+
+    # .score(word): returns the total score value for the given word [breakfast]
+    def self.score(word) # word is input as a string (case insensitive)
+      letters = convert_input(word)
+      letters.inject(0){ |sum, letter| sum + VALUES[letter.to_sym] }
+    end
+
+    # returns the word as an array of lowercase letters, if input is acceptable
+    def self.convert_input(word)
+      acceptable_input_check(word)
+      word.downcase.chars
+    end
+
+    # checks word for not-allowed content. If contains, prompts for new input until accepted.
+    def self.acceptable_input_check(word)
+      # checks for nil, special characters. NOTE: '' is treated as zero, and is an accepted input.
+      status = (
+      if word.is_a?(String) # checks for nil input, arrays, integers (not in a string), etc.
+        word.chars.each do |letter|
+          return false unless ('a'..'z').include?(letter) || ('A'..'Z').include?(letter) #.downcase doesn't work on integers
+        end
+        word.length <= 7
+      end
+      )
+
+      unless status
+        puts "You entered #{word}. This is not accepted input. Please choose a new word (letters only)."
+        new_word = $stdin.gets.chomp # TODO: will need to be re-written when query methods / error message needs change. # Specs written to not prompt in terminal for new input :)
+        status = self.acceptable_input_check(new_word)
```

Comment:
This is good - I like that it will go back through to continue retrieving inputs repeatedly until valid.